sdk/go/common/resource/properties_path_test.go

```diff
@@ -737,7 +737,20 @@ func TestReset(t *testing.T) {
 			nil,
 		},
 		{
-			"Nested array wildcard reset fails",
+			"Nested array wildcard, new array is shorter fails",
```

Comment:
Odd that "Nested array wildcard index, old array is longer fails" was ok but this test picks up the regression?

Reply:
I believe this is because they are fundamentally different paths -- when you are at the end of a path, you go into one branch; when you are not, you take another. In both, we need to check that arrays line up, but in the former, it's to update; in the latter, it's to recurse.